src/simsopt/_core/graph_optimizable.py

```diff
@@ -1295,3 +1315,57 @@ def J(self):
             return self.func(*args, **kwargs)
 
     return TempOptimizable(func, *args, dof_indicators=dof_indicators, **kwargs)
+
+
+class ScaledOptimizable(Optimizable):
+    """
+    Represents an :obj:`~simsopt._core.graph_optimizable.Optimizable`
+    object scaled by a constant factor. This class is useful for
+    including a weight in front of terms in an objective function. For
+    now, this feature works on classes for which ``.J()`` returns an
+    objective value and ``.dJ()`` returns the gradient, e.g. coil
+    optimization.
+
+    Args:
+        factor: (float) The constant scale factor.
+        opt: An :obj:`~simsopt._core.graph_optimizable.Optimizable` object to scale.
+    """
+
+    def __init__(self, factor, opt):
+        self.factor = factor
+        self.opt = opt
+        super().__init__(depends_on=[opt])
+
+    def J(self):
+        return self.factor * self.opt.J()
+
+    @derivative_dec
+    def dJ(self):
```

Comment:
I believe the `partials` argument should not appear here. For examples of `@derivative_dec` usage written by @florianwechsung , see [here](https://github.com/hiddenSymmetries/simsopt/blob/master/src/simsopt/geo/curveobjectives.py#L39), [here](https://github.com/hiddenSymmetries/simsopt/blob/master/src/simsopt/geo/curveobjectives.py#L91), [here](https://github.com/hiddenSymmetries/simsopt/blob/master/src/simsopt/geo/curveobjectives.py#L137), [here](https://github.com/hiddenSymmetries/simsopt/blob/master/src/simsopt/geo/curveobjectives.py#L215). It can be seen that those previous examples of `dJ()` functions do not take a `partials` argument, as the decorator handles that argument.

Reply:
OK.